src/rt/runtime.rs

```diff
@@ -102,74 +88,21 @@ impl Runtime {
     /// Runs the runtime on the current thread.
     pub fn run(&self) {
         scope(|s| {
-            let mut idle = 0;
-            let mut delay = 0;
-
-            loop {
```

Comment:
Why was this loop removed? Doesn't that mean that once we've iterated over all machines once, the loop exits? If this is indeed a bug, we should have a test that catches this.

Reply:
Once we started the task processing thread, we thought there was no runtime work.
Now the scheduler is like that.

https://github.com/async-rs/async-std/blob/2dbebe54ede4d2c0a18380f51e785d5306022ac5/src/task/executor/pool.rs#L31